packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -0,0 +1,734 @@
+const updateModel = (data: Record<string, any>, parent?: string) => {
+  Object.keys(data).forEach(key => {
+    let modelKey = parent ? `${parent}-${key}` : key
+    let scheme = props.schema[modelKey]
+
+    // If `scheme` is undefined, it is either because the key is not the deepest nested key of the schema object
+    // or the field name has dashes in it and its schema key was converted to underscores during initiation.
+    if (!scheme) {
+      const underscoredModelKey = parent ? `${parent}-${key.replace(/-/g, '_')}` : key.replace(/-/g, '_')
+
+      // Here we check if the underscored key exists in the schema and the `fieldNameHasDashes` flag is true
+      if (props.schema[underscoredModelKey]?.fieldNameHasDashes) {
+        modelKey = underscoredModelKey
+        scheme = props.schema[modelKey]
+      }
+    }
+
+    const value = data[key]
+    const type = typeof value
+
+    // Ensure Object Advanced field is saved to model as empty object
+    if (scheme && scheme.type === 'object-advanced' && !value) {
+      formModel[modelKey] = {}
+      originalModel[modelKey] = {}
+
+      return
+    }
+
+    // avoid pass by ref
+    if (Array.isArray(value)) {
+      formModel[modelKey] = JSON.parse(JSON.stringify(value))
+      originalModel[modelKey] = JSON.parse(JSON.stringify(value))
+
+      return
+    }
+
+    // update the model for each child of the object
+    if (type === 'object' && value && !value.length) {
+      return updateModel(value, modelKey)
+    }
+
+    // treatment for arrays, convert to comma separated string
+    if (type === 'object' && value && value.length && scheme.type === 'input') {
+      formModel[modelKey] = value.join(', ')
+      originalModel[modelKey] = value.join(', ')
+
+      return
+    }
+
+    formModel[modelKey] = value
+    originalModel[modelKey] = value
+  })
+
+  emit('model-updated', {
+    model: formModel,
+    originalModel,
+    data: getModel(),
+  })
+}
+
+const loading = ref(true)
+const initFormModel = (): void => {
+  if (props.record && props.schema) {
+    // global fields
+    updateModel({
+      enabled: props.record.enabled ?? true,
+      ...(props.record.instance_name && { instance_name: props.record.instance_name || '' }),
+      ...(props.record.protocols && { protocols: props.record.protocols }),
+      ...(props.record.tags && { tags: props.record.tags }),
+    })
+
+    if (props.record.data) {
+      updateModel(props.record.data)
+    } else if (props.record.config) {
+      // scope and top level fields
+      if (props.record.consumer_id || props.record.service_id || props.record.route_id || props.record.consumer_group_id) {
+        updateModel({
+          service_id: props.record.service_id,
+          route_id: props.record.route_id,
+          consumer_id: props.record.consumer_id,
+          consumer_group_id: props.record.consumer_group_id,
+          enabled: props.record.enabled,
+        })
+      }
```

Comment:
When editing a plugin that is scoped to both a service and a route, the scope input is blank. Not sure if the following is the right fix, but it does avoid the blank input:
```suggestion
      if (props.record.consumer?.id || props.record.service?.id || props.record.route?.id || props.record.consumer_group?.id) {
        updateModel({
          'service-id': props.record.service?.id,
          'route-id': props.record.route?.id,
          'consumer-id': props.record.consumer?.id,
          'consumer_group-id': props.record.consumer_group?.id,
          enabled: props.record.enabled,
        })
      }
```